### PR TITLE
Houdini: Implement Arnold .ass standin extraction from Houdini (also support .ass.gz)

### PR DESCRIPTION
**Pull request by BigRoy**
## Brief description

This implements the creation and publishing of Arnold proxies (`.ass` or `.ass.gz`) from Houdini.

Currently defaults to ~`.ass.gz`~ (changed with https://github.com/pypeclub/OpenPype/pull/2603/commits/dd92fe1e4ee9a195b85d7aa6b4234fba2829dec4) `.ass`
#### Explicit detection of multidot extension `.ass.gz`

Detecting whether a filename ends with a multidot extension like `.ass.gz` isn't fool proof. It's hard to know whether something is just an extra dot in a filename or actually is part of the extension. As such I've opted for logic that explicitly says that it allows `.ass.gz` as extension.

It does mean some hardcoded logic for that - but will definitely reduce false positives. If anyone knows a better way I'd be happy to hear it.

#### Both `.ass.gz` and `.ass` extensions publish as `ass` representation

No matter whether you publish `.ass` or `.ass.gz` extensions the resulting representation name is `ass`. This just allows for a better versioning/updating experience for the end user so that to update from a version that was `.ass` and a new version being `.ass.gz` that you can actually update through the manager - otherwise using Avalon's `switch` logic would be required to do so.

## Testing notes:

The simplest setup:
```python
import hou
import avalon.api
import openpype.lib

# Create a Box so we have something to export
root = hou.node("/obj")
geo = root.createNode("geo")
box = geo.createNode("box")

# Create Arnold ASS instance
Creator = openpype.lib.get_creator_by_name("CreateArnoldAss")
asset = avalon.api.Session["AVALON_ASSET"]
container = avalon.api.create(Creator, 
                              name="assMain", 
                              asset=asset)

# Publish!
```

Basically create a Box, create the ASS ROP and publish. The Arnold ROP acts like a render rop and "extracts" whatever is set to display in the viewport by default (using `Candidate Objects = *` on the ROP). Since the Box is visible that would end up in the publish.

This requires `htoa` (Houdini to Arnold) to be installed for Houdini.

### CODE REVIEW

**Comment by BigRoy** [comment]
I am going to change the default to `.ass` instead of `.ass.gz`. We are hitting cases where the gzipped ass files are bigger somehow. Plus writing them is tons slower than writing regular `.ass`.

Will also likely expose a toggle in settings to set the default.

**Comment by BigRoy** [comment]
> I am going to change the default to .ass instead of .ass.gz. We are hitting cases where the gzipped ass files are bigger somehow. Plus writing them is tons slower than writing regular .ass.
> 
> Will also likely expose a toggle in settings to set the default.

Implemented. ✅

**Comment by antirotor** [comment]
It doesn't work for me for some reason.

image

```
Traceback (most recent call last):
  File "E:\projects\openpype\3x\.venv\lib\site-packages\pyblish\plugin.py", line 522, in __explicit_process
    runner(*args)
  File "E:\projects\openpype\3x\openpype\hosts\houdini\plugins\publish\collect_frames.py", line 28, in process
  File "E:\projects\openpype\3x\openpype\hosts\houdini\api\lib.py", line 184, in get_output_parameter
    if node.evalParm("ar_ass_export_enable"):
TypeError: Node type 'arnold' not supported
```
even if I have the arnold up and running.

Also, shouldn't the creator respect **Use selection** checkbox?

**Comment by BigRoy** [comment]
@antirotor that error message looks like you were running using older python code somehow and updated in between. That error message wouldn't happen on that particular line of code? Can you try again, checkout, restart tray and Houdini and give it a whirl. Maybe ditch any pycache you have too just to be sure.

The code is definitely in the list of changes that should make that work.

**Comment by BigRoy** [comment]
> That error message wouldn't happen on that particular line of code? Can you try again, checkout, restart tray and Houdini and give it a whirl. Maybe ditch any pycache you have too just to be sure.

@antirotor could you check again?

**Review by antirotor — APPROVED** [review]
Its working well. After some time, Integrate new becomes strangle slow, problably unrelated to this PR.
